### PR TITLE
Add test benchmark

### PR DESCRIPTION
**Pull request by xye16**
Add a test benchmark that supports:

1. Use either random generated data (`eval_with_random_input`) or string-like input (`eval_with_string_data`) as the input
2. Comparing the outputs between two functions with the same input
3. Measuring the runtime

### CODE REVIEW

File: brute_force.cpp [hunk 1]
```diff
@@ -1,18 +1,14 @@
...
-static inline char get(const char *arr, int idx) {
-    return (arr[idx>>2] >> ((idx & 0x3) << 1)) & 0x3;
-}
+#include "char_compress.hpp"
 
 int brute_force_search(
-    const char *text, int text_length, const char *pattern, int pattern_length,
-    int *output, int max_output_cnt, int *fail
+    const char *text, int text_length, const char *pattern, int16_t pattern_length,
+    int16_t *output, int max_output_cnt, int16_t *fail
```
Comment (aryanraja1072) [inline brute_force.cpp:6]: If we are storing indices in text data in `*output`, will we not overflow if we use int16_t?

**Review by aryanraja1072 — COMMENTED** [review]
.

**Review by aryanraja1072 — COMMENTED** [review]
Please confirm the choice of data type for output. Rest looks good to me.

File: brute_force.cpp (hunk 1 above)
Comment (xye16) [reply brute_force.cpp]: You're right. Changed to int.

**Pull request merged**